ccl: add Cockroach Community License text

danhhz opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/13007)
<!-- Reviewable:end -->

danhhz committed:
ccl: add Cockroach Community License text

bdarnell commented:
We should also rename the top-level LICENSE file to something like APL.txt and add a new top-level LICENSE file explaining that some packages are CCL and some are APL.

danhhz committed:
cockroach: describe the two LICENSEs

Move the Apache Licence text to APL.txt and use the top level LICENSE
file to point at the two licenses.

danhhz commented:
@bdarnell, something like this?

---

Review status: 0 of 3 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13007#-:-KasR-aPGC1BXNLYKMjm:b-2c8djd)*
<!-- Sent from Reviewable.io -->

dt commented on pkg/ccl/LICENSE:
```diff
@@ -0,0 +1,419 @@
+    (c) Record-keeping and Audit.  If fees for CockroachDB Enterprise Edition
+        are based on the number of cores or servers running on CockroachDB
+        Enterprise Edition or another use-based unit of measurement, You must
+        maintain complete and accurate records with respect Your use of
```
is this supposed to be "with respect *to* Your use..."?

bdarnell commented on LICENSE:
```diff
@@ -1,202 +1,10 @@
-                           Version 2.0, January 2004
-                        http://www.apache.org/licenses/
+Each of the CockroachDB source files is covered by one of two licences. Files
+under the ccl/ folder are covered by the
```
`pkg/ccl`, not just `ccl/`, right?

danhhz commented:
Review status: 0 of 3 files reviewed at latest revision, 2 unresolved discussions.

---

*[LICENSE, line 2 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/13007#-KasQzSH3-6j9QBoqlut-r2-2:-KasUi2NBVYiUfAhd7MK:btnprka) ([raw file](https://github.com/cockroachdb/cockroach/blob/883063f89bca181c7e51aff0b7c8897c7129c38c/LICENSE#L2)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary><blockquote>

`pkg/ccl`, not just `ccl/`, right?
</blockquote></details>

Deleted these entirely so they have no chance to rot

---

*[pkg/ccl/LICENSE, line 189 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/13007#-KasIvs0TzEtSmkRwecS-r1-189:-KasUhH8OzXxmRqF5raB:bb6un1j) ([raw file](https://github.com/cockroachdb/cockroach/blob/2efac31a0fc324177a517afc63d60fbafb81df03/pkg/ccl/LICENSE#L189)):*
<details><summary><i>Previously, dt (David Taylor) wrote…</i></summary><blockquote>

is this supposed to be "with respect *to* Your use..."?
</blockquote></details>

Done

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13007)*
<!-- Sent from Reviewable.io -->

bdarnell commented:
Review status: 0 of 3 files reviewed at latest revision, 2 unresolved discussions.

---

*[LICENSE, line 2 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/13007#-KasQzSH3-6j9QBoqlut-r2-2:-KasVOu4kpt8tIGJJj9d:b-pdr3ji) ([raw file](https://github.com/cockroachdb/cockroach/blob/883063f89bca181c7e51aff0b7c8897c7129c38c/LICENSE#L2)):*
<details><summary><i>Previously, danhhz (Daniel Harrison) wrote…</i></summary><blockquote>

Deleted these entirely so they have no chance to rot
</blockquote></details>

I think it's good to get rid of the specification about which files are under which license, but we still need to point to the license text here (especially since the CCL is not in an obvious location. Maybe it should be moved up to top level instead of inside pkg/ccl?)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13007)*
<!-- Sent from Reviewable.io -->

danhhz commented:
Review status: 0 of 3 files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

*[LICENSE, line 2 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/13007#-KasQzSH3-6j9QBoqlut-r2-2:-Kas_bR4fByTCiAN1EXa:b-oxjbc9) ([raw file](https://github.com/cockroachdb/cockroach/blob/883063f89bca181c7e51aff0b7c8897c7129c38c/LICENSE#L2)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary><blockquote>

I think it's good to get rid of the specification about which files are under which license, but we still need to point to the license text here (especially since the CCL is not in an obvious location. Maybe it should be moved up to top level instead of inside pkg/ccl?)
</blockquote></details>

Added the paths back. I'd like to leave the ccl license where it is for now, if that's okay

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13007)*
<!-- Sent from Reviewable.io -->

danhhz commented:
Thanks for the reviews!
